src/components/page-item.tsx

```
@@ -0,0 +1,56 @@
+/**
+ * Renders a table row for a transcript page with editable start/end times, text, and selection controls.
+ *
+ * Allows users to select the page, edit its timing and text, and interact with the text area for advanced behaviors such as right-to-left input, auto-resizing, and custom paste handling for Arabic text.
+ *
+ * @param page - The transcript page to display and edit.
+ */
+const PageItem = ({ page }: { page: Page }) => {
+    const { toggleSegmentSelection } = useTranscriptStore.getInitialState();
+    const isSelected = useManuscriptStore((state) => state.selectedPages.includes(page));
+    const lineHighlights: { [lineNumber: number]: string } = page.errorLines?.reduce(
+        (acc, e) => ({ ...acc, [e]: 'bg-red-100' }),
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)